### ISSUE TITLE
Difference between Export-M365DSCConfiguration resources AADGroup and O365Group?

### ISSUE THREAD

**Issue by muellmk** [microsoft/Microsoft365DSC#3100]
Can someone explain the difference between the Export-M365DSCConfiguration resources "AADGroup" and "O365Group"?

I have 22 groups in my tenant.  Some are M365 groups and other are Security groups.

When I run **Export-M365DSCConfiguration -Credential $cred -Components @("AADGroup") -Mode Full** as global admin, I get all 22 groups returned in the export.

When I run **Export-M365DSCConfiguration -Credential $cred -Components @("O365Group") -Mode Full** as global admin, I only get all 20 groups returned.  I cannot see anything unique about the two groups not returned that would prevent them from being included in the export.

**Comment by NikCharlebois** [comment]
There are quite a lot of similarities between the two and we are considering merging them into a single resource in a future release. Essentially, the differences are:

* **AADGroup**: Only gets groups that:
  * Aren't both Security & Mail Enabled
  * Aren't mail enabled and don't have an assigned group type.

* **O365Group**: Only get groups that:
  * Don't have their MailNickName property set to '00000000-0000-0000-0000-000000000000'

**Comment by NikCharlebois** [comment]
Essentially:


* AADGroup:
```
$groups = Get-MgGroup -Filter $Filter -All:$true -ErrorAction Stop
        $groups = $groups | Where-Object -FilterScript {-not ($_.MailEnabled -and ($null -eq $_.GroupTypes -or $_.GroupTypes.Length -eq 0)) -and
                                                        -not ($_.MailEnabled -and $_.SecurityEnabled)}
```

* O365Group: 
```
Get-MgGroup -All:$true -Filter $Filter | Where-Object -FilterScript {
            $_.MailNickName -ne '00000000-0000-0000-0000-000000000000'
        }
```

**Comment by muellmk** [comment]
Nik,

Thanks for the details on how the groups are queried. 

One suggestion I have for O365Group would be to restrict the groups returned to only those listed on the Microsoft 365 tab of the M365 Admin Center active groups page.  By definition, I would expect O365Group to only return groups of type "Microsoft 365".  This would make for a like-for-like comparison between the exported results and what is display in the Admin Center.

Alternatively, if you do end up with a single "group" resource, simply export all groups with enough properties to allow filtering exported results.

Marc

**Comment by icpw** [comment]
> Nik,
> 
> Thanks for the details on how the groups are queried.
> 
> One suggestion I have for O365Group would be to restrict the groups returned to only those listed on the Microsoft 365 tab of the M365 Admin Center active groups page. By definition, I would expect O365Group to only return groups of type "Microsoft 365". This would make for a like-for-like comparison between the exported results and what is display in the Admin Center.
> 
> Alternatively, if you do end up with a single "group" resource, simply export all groups with enough properties to allow filtering exported results.
> 
> Marc

My vote goes with the alternative.  The group type is important.  Any Dynamic rule definition is important. Exchange distribution lists (static and dynamic) should be included.

**Comment by FabienTschanz** [comment]
Is this still a present issue?

**Comment by Borgquite** [comment]
@FabienTschanz Yes - we're still exporting groups twice. Could we just get rid of O365Group? I'm not really sure what it adds when AADGroup should cover everything you can manage with Graph:

- Entra assigned security groups (including role assignment groups)
- Entra dynamic security groups
- Microsoft 365 assigned groups (including Teams Enabled groups)
- Microsoft 365 dynamic groups (including Teams Enabled groups)

And EXOGroupSettings and EXODistributionGroup cover everything you can do in Exchange:
- Exchange settings for Microsoft 365 groups
- Distribution Lists
- Mail-enabled security groups

The only group type we're missing is dynamic Exchange distribution groups, which I know you're working on...

?
